### ISSUE TITLE
Question about something with setBreakPointsRequest

### ISSUE THREAD

**Issue by StephenBloomquist** [harold-b/vscode-duktape-debug#18]
Is doRemoveBreakpoints accounting for the fact that the duktape breakpoint indices are going to shift after the request is processed? If so, how? I can't quite work it out.

**Comment by harold-b** [comment]
I don't currently recall the inner workings of the code off the top of my head, sorry. I'm preoccupied with a project deliverable at the moment so I can't get into it right now either, so you'll have to dig in  :rice: :)

**Comment by StephenBloomquist** [comment]
Yeah I've already looked it into a bit. It doesn't appear to be handling it. From what I can work out `getBreakpointsForFile` is returning an array of breakpoints with the indices in ascending order. They're getting added to the `remBPs` array in ascending order as well. 

If kept this way `doRemoveBreakpoints` will have to shift all the `dukIdx` values down every time it calls `requestRemoveBreakpoint` to keep them in-line with duktape (If you remove index 5, then the breakpoint you have at index 8 would now be at 7, but duktape is being asked to remove at index 8).

I think a better way would be to have the `remBPs` array sorted such that the `dukIdx` are in descending order, because then you don't have to care that all the indices above the index you just removed got shifted.

**Comment by harold-b** [comment]
If I recall, the packed index were accounted for. I can't quite recall specifics, so I'll have a look when I get the chance to refactor.

**Comment by StephenBloomquist** [comment]
It _is_ accounted for in `removeAllTargetBreakpoints` but if it's accounted for in `setBreakPointsRequest` then I've misread some logic somewhere.

**Comment by harold-b** [comment]
Thanks, I'll also have a look when I get the chance to dive back in :+1:
